contrib/bind/provider.py

```diff
@@ -43,8 +43,8 @@ def remove_record(self, record):
 
     def first_setup(self, parser):
         # Insert SOA and NS records
-        hostname = unit_get('public-address')
-        addr = resolve_hostname_to_ip(hostname)
+        addr = unit_get('public-address')
+        #addr = resolve_hostname_to_ip(hostname)
```

Comment:
Ok which substrate was this tested on? i'd like to run a series of validations across the clouds so we can set sail to that method then. Last i checked AWS was still giving the unit name, as well as single interface MAAS nodes. 

We can feesibly update that method to check with a regex, compare ipv4 vs alphanumeric to determine if we need to pass the data through the handler. Which seems like a slightly better resolution that maintains compatibility in the cases where it would fail.